### PR TITLE
Convert tabs bar with Aphrodite

### LINKED ISSUE #11272: Convert tabs bar with Aphrodite to remove tabs.less
### Description

Refactor tabs bar to remove `tabs.less`.

### Additional Information

It would be appreciated if #10686 would be based on this task. CC @petemill

### PR DESCRIPTION
**Pull request by luixxiul**
Closes #11272

- Remove tabs.less
- Set color properties on theme.js
- Ser default properties on tabIcon.js
- BrowserButton component is applied to back/forward button and menu button on the tabs bar

Also:
- Apply no-drag to the new tab button and tab area
- Make the default icon large
- Add will-change:opacity to the animations
- Remove 'content' from theme.tab on theme.js

Test Plan: `CXX=g++-4.8 NODE_ENV=test TEST_DIR=tab-components npm run testsuite`

For QA team:

Test Plan:
1. Test that tabs are pinnable
2. Test that tabs are unpinnable
3. Test that tabs are draggable to same tabset
4. Test that tabs are draggable to alternate tabset
5. Test that tabs can be teared off into a new window
6. Test that you are able to reattach a tab that is teared off into a new window
7. Test that tab pages can be closed
8. Test that tab pages can be muted

Test Plan:
1. Test that the area around tab page indicators is draggable
2. Test that the gap between them is draggable

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/renderer/components/tabs/content/closeTabIcon.js [hunk 1]
```diff
@@ -17,8 +17,7 @@ const frameStateUtil = require('../../../../../js/state/frameStateUtil')
 
 // Styles
 const {theme} = require('../../styles/theme')
-const {spacing, zindex} = require('../../styles/global')
-const {opacityIncreaseKeyframes} = require('../../styles/animations')
+const globalStyles = require('../../styles/global')
```
Comment (luixxiul) [inline app/renderer/components/tabs/content/closeTabIcon.js]: Before doing that task, let's open an issue to edit [this section on docs/style.md](https://github.com/brave/browser-laptop/blob/master/docs/style.md#where-to-put-my-code) to achieve a common understanding.

File: app/renderer/components/tabs/content/closeTabIcon.js [hunk 2]
```diff
@@ -66,37 +67,33 @@ class CloseTabIcon extends React.Component {
...
-    animationName: opacityIncreaseKeyframes,
-    animationTimingFunction: 'linear',
-    animationDuration: '200ms',
-    animationDelay: '25ms',
-    animationFillMode: 'forwards',
-
+  icon_closeTab: {
     boxSizing: 'border-box',
-    zIndex: zindex.zindexTabsThumbnail,
+    zIndex: globalStyles.zindex.zindexTabsThumbnail,
```
Comment (luixxiul) [inline app/renderer/components/tabs/content/closeTabIcon.js]: Ditto.

File: app/renderer/components/tabs/tabPages.js [hunk 3]
```diff
@@ -22,7 +24,7 @@ class TabPages extends React.Component {
   }
 
   render () {
-    return <div className='tabPageWrap'>
+    return <div className={css(styles.tabPageWrap)}>
```
Comment (luixxiul) [inline app/renderer/components/tabs/tabPages.js]: It'll be renamed into `tabPages` (`Wrap` is not required with BEM naming)

**Comment by luixxiul** [comment]
@petemill would you please review the PR?

File: app/renderer/components/tabs/tabPage.js [hunk 4]
```diff
@@ -144,19 +147,54 @@ class TabPage extends React.Component {
...
+  tabPage_audio: {
+    border: `1px solid ${theme.audio.color}`
+  },
+
+  tabPage_active: {
+    backgroundColor: theme.tabPage.active.backgroundColor,
+    border: 'none',
+
+    ':hover': {
+      border: `1px solid ${theme.tabPage.active.hover.borderColor}`
```
Comment (cezaraugusto) [inline app/renderer/components/tabs/tabPage.js]: active tab page should have the same state when hovering (filled orange)

Comment (luixxiul) [reply app/renderer/components/tabs/tabPage.js]: On theme.js, they are set like this:

````
    tabPage: {
      backgroundColor: '#fff',
      borderColor: '#bbb',

      hover: {
        borderColor: globalStyles.color.braveOrange
      },

      active: {
        backgroundColor: globalStyles.color.braveOrange,

        hover: {
          borderColor: globalStyles.color.braveOrange
        }
      }
````

So they should be same by default. Is it OK?

Comment (cezaraugusto) [reply app/renderer/components/tabs/tabPage.js]: not sure what is happening but this is what I'm seeing now:

nov-03-2017 23-53-46

Comment (cezaraugusto) [reply app/renderer/components/tabs/tabPage.js]: maybe you should enforce the background color on hover for the active tab page too

Comment (luixxiul) [reply app/renderer/components/tabs/tabPage.js]: Wow nice find! I addressed that here: https://github.com/brave/browser-laptop/pull/11530/commits/0582859ef29a0c5c0589b972fff17a31545543cf#diff-b39b5cd9b6e26e313c7387108e17891cR198

**Comment by luixxiul** [comment]
@cezaraugusto thanks for the review. I appreciate that very much ❤️ 😄

**Comment by petemill** [comment]
@luixxiul apologies for the delayed review. I was scared from all the changes I'll have to make to the PR which deals which new tab styles #11720! But thanks for putting the work in, and I look forward to merging with it.

**Comment by petemill** [comment]
I'll have to move this milestone up to 0.22 (from 0.23) since #12565 is for 0.22 and is completely based on this refactor. Attempting merge to see if this is huge or not...

**Comment by petemill** [comment]
0.22.x 50f033f293af2819f772fafd38e740dc6d258480

**Pull request merged**